mtags/src/main/scala-2/scala/meta/internal/pc/InferredMethodProvider.scala

```diff
@@ -0,0 +1,520 @@
+              case _ =>
+                // object Y {}
+                // Y.nonExistent(1,2,3)
+                typedTreeAt(containerSymbol.symbol.pos) match {
+                  // class case
+                  case ClassDef(
+                        _,
+                        _,
+                        _,
+                        template
+                      ) =>
+                    makeClassMethodEdits(template)
+                  // trait case
+                  case ModuleDef(
+                        _,
+                        _,
+                        template
+                      ) =>
+                    makeClassMethodEdits(template)
+                  case _ =>
+                    unimplemented("object-method-inner")
+                }
+            }
+          case _ =>
+            unimplemented("object-method-outer")
+        }
+      } else
+        unimplemented("object-method")
+    }
+    val uriTextEditPairs = typedTree match {
+      case errorMethod: Ident if errorMethod.isErroneous =>
+        lastVisitedParentTrees match {
+          /**
+           * Works for apply with unknown name:
+           * ```scala
+           * method(nonExistent(param))
+           * ```
+           */
+          case _ :: (nonExistent: Apply) :: Apply(
+                Ident(containingMethod),
+                arguments
+              ) :: _ if nonExistent.isErrorTyped =>
+            makeEditsForApplyWithUnknownName(
+              arguments,
+              containingMethod,
+              errorMethod,
+              nonExistent
+            )
+          // <<nonExistent>>(param1, param2)
+          // val a: Int = <<nonExistent>>(param1, param2)
+          case (_: Ident) :: Apply(
+                containing @ Ident(
+                  nonExistent
+                ),
+                arguments
+              ) :: _ if containing.isErrorTyped =>
+            signature(
+              name = nonExistent.toString(),
+              paramsString = Option(argumentsString(arguments).getOrElse("")),
+              retType = None,
+              postProcess = identity,
+              position = None
+            )
+          // List(1, 2, 3).map(<<nonExistent>>)
+          // List((1, 2)).map{case (x,y) => <<nonExistent>>(x,y)}
+          case (_: Ident) :: Apply(
+                Ident(containingMethod),
+                arguments
+              ) :: _ =>
+            makeEditsForListApply(arguments, containingMethod, errorMethod)
+          // List(1,2,3).map(myFn)
+          case (_: Ident) :: Apply(
+                Select(
+                  Apply(
+                    Ident(_),
+                    arguments
+                  ),
+                  _
+                ),
+                _
+              ) :: _ =>
+            signature(
+              name = errorMethod.name.toString(),
+              paramsString =
+                Option(argumentsString(arguments.take(1)).getOrElse("")),
```

Comment:
What if we have a method like:

```
object DoubleList{
  def apply(a : Int) = List( (a,b) )
}

class DoubList{
  def map(f: (Int, Int) => String) = ???
}


```

`DoubleList(1).map(nonExistent)` would create a wrong method. We can't know what is the type used in a method based on the arguments of the preceding apply.

Reply:
A very hacky way around it would be to create a function that take in everything like
`def myErrorFunction(a: Any) : Nothing` and with that rerun the presentation compiler to get the actual type of map. We do a similar thing with updating inferred type : https://github.com/scalameta/metals/blob/main/mtags/src/main/scala-2/scala/meta/internal/pc/InferredTypeProvider.scala#L89C9-L89C26

But maybe we could do it later.

Reply:
ignored the cases, thanks